plutus-ir/src/Language/PlutusIR/Value.hs

```diff
@@ -8,15 +8,15 @@ isTermValue :: Term tyname name a -> Bool
 isTermValue = \case
     -- Let is not a value (will compile into applications and/or type instantiations)
     Let {} -> False
-    -- Lambdas and constants are always values
+    -- Lambdas, variables, constants, and builtins are always values
     LamAbs {} -> True
+    Var {} -> True
     Constant {} -> True
     Builtin {} -> True
     -- Type abstractions and wraps are values if their bodies are
     TyAbs _ _ _ t -> isTermValue t
     Wrap _ _ _ t -> isTermValue t
     -- All other PLC terms are not values
-    Var {} -> False
     Apply {} -> False
     TyInst {} -> False
```

Comment:
So is `addInteger {integer 1} 1` not a value?

Reply:
Rats, this is missing the whole builtin-applications-as-values workaround. I'm not sure what I was thinking! Maybe I thought that were about to move away from that... Will fix.

Reply:
I think, it's a non-trivial thing actually, because you have to distinguish between `addInteger {integer 1} 1` and `addInteger {integer 1} 1 2`, because the former is in NF while the latter isn't.

Reply:
The situation here is weird. Let's ask the question: should a saturated builtin be a value?

- The current spec says no, they are terms
    - This makes sense with them not being normal forms
- The current implementation in `language-plutus-core` says nothing is a `builtinValue` because it uses `Constant` instead of `Builtin` to discriminate the base case, but if that bug were fixed it would say even saturated builtin applications were values.
    - This is inline with the old spec.

I think I'm going to follow the old spec for now since we haven't changed the builtin handling yet, and just make all applications of builtins into values.

Reply:
Sounds good to me.

> Let's ask the question: should a saturated builtin be a value?

I believe, no. As per the current spec.